File: Ruby/ruby/RPG/human.rb
class Human 
    attr_accessor :strength, :intelligence, :stealth, :health 
    def initialize
        @strength = 3 
        @intelligence = 3 
        @stealth = 3 
        @health = 100 
    end 

    def attack(target)
        if target.class.ancestors.include?(Human)
            puts "You attacked #{target}"
            target.health -= 10
            true
        else  
            puts "Invalid target"
            false
        end 
    end 
end

human1 = Human.new 
human2 = Human.new 
human1.attack(human2)
human1.attack("Not a Human")
puts human2.health 
puts human1.health 

File: Java/Spring/ObjectMaster/src/com/objectmaster/models/Human.java
package com.objectmaster.models;

public class Human {
	public int strength;
	public int stealth;
	public int intelligence;
	public int health;
	
	public Human() {
		this.strength = 3;
		this.stealth = 3;
		this.intelligence = 3;
		this.health = 100;
	}
	
	public void Attack(Human target) {
		
		target.health -= this.strength;
	}
	
}
